src/sshfs_mount/sshfs_server.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2018 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <QStringList>
+
+#include "../ssh/ssh_client_key_provider.h" // FIXME
+#include <multipass/exceptions/sshfs_missing_error.h>
+#include <multipass/logging/log.h>
+#include <multipass/logging/standard_logger.h>
+#include <multipass/ssh/ssh_session.h>
+#include <multipass/sshfs_mount/sshfs_mount.h>
+
+#include <signal.h>
+#include <sys/prctl.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+using namespace std;
+
+namespace
+{
+unordered_map<int, int> deserialise_id_map(const char* in)
+{
+    unordered_map<int, int> id_map;
+    QString input(in);
+    auto maps = input.split(',', QString::SkipEmptyParts);
+    for (auto map : maps)
+    {
+        auto ids = map.split(":");
+        if (ids.count() != 2)
+        {
+            cerr << "Incorrect ID mapping syntax";
+            continue;
+        }
+        bool ok1, ok2;
+        int from = ids.first().toInt(&ok1);
+        int to = ids.last().toInt(&ok2);
+        if (!ok1 || !ok2)
+        {
+            cerr << "Incorrect ID mapping ids found, ignored" << endl;
+            continue;
+        }
+        id_map.insert({from, to});
+    }
+    return id_map;
+}
+} // namespace
+
+int main(int argc, char* argv[])
+{
+    prctl(PR_SET_PDEATHSIG, SIGHUP); // ensure if parent dies, this process gets the SIGHUP signal
```

Comment:
This certainly won't work in Windows, so this should be platform specific and hopefully there is an equivalent Windows way to do this.

Reply:
Correct, I intend just to ifdef these bits away. Abstraction layer not useful as the actual implementation of this behaviour is quite different on the various platforms.

Reply:
I'd really like to stay away from ifdef'ing this up.  Instead, I'd rather see something like `mp::platform::handle_parent_dying()` (or whatever) and then implement it in the various platform files. Is something like that possible?